code/modules/modular_computers/file_system/programs/ntdownloader.dm

```diff
@@ -99,10 +104,16 @@
 			download_netspeed = NTNETSPEED_HIGHSIGNAL
 		if(NTNET_ETHERNET_SIGNAL)
 			download_netspeed = NTNETSPEED_ETHERNET
-	if(download_netspeed)
-		if(HAS_TRAIT(computer, TRAIT_MODPC_HALVED_DOWNLOAD_SPEED))
-			download_netspeed *= 0.5
-		download_completion += download_netspeed
+	if(download_netspeed <= 0)
+		return
+	if(HAS_TRAIT(computer, TRAIT_MODPC_HALVED_DOWNLOAD_SPEED))
+		download_netspeed *= 0.5
+
+	download_completion += download_netspeed
+	if(download_completion > downloaded_file.size)
+		// We don't complete it here so we stay on 100% for a cycle
+		// We do cap out our completion to avoid UI issues
+		download_completion = downloaded_file.size
```

Comment:
```suggestion
	if(HAS_TRAIT(computer, TRAIT_MODPC_HALVED_DOWNLOAD_SPEED))
		download_netspeed *= 0.5
	// We don't complete it here so we stay on 100% for a cycle
	// We do cap out our completion to avoid UI issues
	download_completion = min(download_completion  + download_netspeed, downloaded_file.size)
```